Sources/Sentry/SentryFrameInAppLogic.m

```diff
@@ -0,0 +1,44 @@
+#import "SentryFrameInAppLogic.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface
+SentryFrameInAppLogic ()
+
+@property (nonatomic, copy, readonly) NSArray<NSString *> *inAppIncludes;
+@property (nonatomic, copy, readonly) NSArray<NSString *> *inAppExcludes;
+
+@end
+
+@implementation SentryFrameInAppLogic
+
+- (instancetype)initWithInAppIncludes:(NSArray<NSString *> *)inAppIncludes
+                        inAppExcludes:(NSArray<NSString *> *)inAppExcludes
+{
+    if (self = [super init]) {
+        _inAppIncludes = inAppIncludes;
+        _inAppExcludes = inAppExcludes;
+    }
+
+    return self;
+}
+
+- (BOOL)isInApp:(NSString *)imageName
+{
+    for (NSString *inAppInclude in self.inAppIncludes) {
+        if ([imageName.lastPathComponent.lowercaseString hasPrefix:inAppInclude.lowercaseString])
+            return YES;
+    }
+
+    for (NSString *inAppExlude in self.inAppExcludes) {
+        if ([imageName.lastPathComponent.lowercaseString hasPrefix:inAppExlude.lowercaseString])
+            return NO;
+    }
```

Comment:
L: more like a double-check that `hasPrefix` would work in every case, looks like on iOS is paths instead of packages like Java, so would it make sense to be a sort of `contains`?

Reply:
In Java we use [`startsWith`](https://github.com/getsentry/sentry-java/blob/4b8b7d7d11e216153ec7fd2619490894db02ef7e/sentry/src/main/java/io/sentry/SentryStackTraceFactory.java#L81-L88). Also the [docs state](https://docs.sentry.io/platforms/android/configuration/options/#in-app-include) it's a prefix, so I think we should stay with `hasPrefix`.

Reply:
because on Java it is packages, as stated before, on iOS is different, its not a prefix by concept already because you get the `imageName.lastPathComponent` anyway, what I mean is, is `lastPathComponent` the only important thing here? if it is, then `hasPrefix` should be enough

Reply:
Yes, the `lastPathComponent` is the only important thing here.